grype/pkg/package_test.go

```diff
@@ -0,0 +1,229 @@
+package pkg
+
+import (
+	"testing"
+
+	"github.com/scylladb/go-set/strset"
+
+	"github.com/anchore/syft/syft/file"
+	syftPkg "github.com/anchore/syft/syft/pkg"
+	"github.com/scylladb/go-set"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPackageMetadataExtraction(t *testing.T) {
```

Comment:
nit: I might consider a rename to the convention of naming the function under test explicitly, optionally suffixed with the kind of testing being done, e.g. something like `TestNew_MetadataExtraction`.